substrate/client/transaction-pool/src/fork_aware_txpool/fork_aware_txpool.rs

```diff
@@ -469,12 +469,59 @@ where
 		self.view_store.futures_at(at)
 	}
 
+	/// Searches in the view store for a view by iterating through the fork of
+	/// the `at` block, up to the last finalized block.
+	///
+	/// Returns with a pair of a maybe view and a set of enacted blocks (possibly empty if no view
+	/// is found), usually starting at the last known block of the fork, up to the block
+	/// corresponding to the found view.
+	fn find_best_view(
```

Comment:
In my mental model best view is the view which is the base during handling new block. It is the view the we clone from. I'd prefer to stick to this naming to avoid confusion.
So I'd propose to re-name this method, maybe:
```suggestion
	fn find_view_descendant_upto_hash(
```
or just `find_view_descendant`.

Reply:
This method could also be moved to the `ViewStore` (assuming we pass the hash).

Reply:
Did it here: [9d47e15](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/9d47e15417b4cf3fe54ec942fc883f852e494b32) + [5d7c1fb](https://github.com/paritytech/polkadot-sdk/pull/8533/commits/5d7c1fb93a339fe98078354f57e28b676ce44de7)